### ISSUE TITLE
res.jsonx returns "text/html" even if "res.wantsJson"

### ISSUE THREAD

**Issue by brandonhesse** [balderdashy/sails#2855]
A noticeable error coming from the recent version of 0.11.0:

A default fresh app, when using proper `Accept: application/json` headers in Postman, the returned data is still a text/html. Reviewing the code and setting breakpoints, `req.wantsJson()` is correctly calling `res.jsonx()`, however at line [213](https://github.com/balderdashy/sails/blob/master/lib/hooks/responses/index.js#L213) of index.js in the lib/hooks/responses the return is a `res.send()`. While I'm sure this is intended behavior for the `res.jsonx()`, I do not believe it was intended for all the api/responses that come default to return the raw string without a object wrapper as found [here](https://github.com/balderdashy/sails-generate-backend/blob/master/templates/api/responses/notFound.js#L45)

The obvious solution is to modify these files to wrap the `data` in a `{ data: data }` before sending it to `res.jsonx()`, but I think I should issue a pull-request to the [generator](https://github.com/balderdashy/sails-generate-backend).

**Comment by CWyrtzen** [comment]
@brandonhesse is this still an issue? Did you issue the PR?

**Comment by brandonhesse** [comment]
Still an issue, and I did not make a PR. I'm not to familiar with making PRs and was more concerned with reporting the bug. I wasn't even sure if this was the right place to report this, as it's not really an issue with the sails.js framework's internals, but a pre-generated template from it's new project command line.

Because I wasn't responded to back in April I was unsure if I was in the right place, and never came back to it because I was trying out serval frameworks awaiting at least someone to point me to the right direction.

**Comment by adamwaite** [comment]
Hi guys, I have just run into this issue too:

```
describe.only('Unauthenticated (no token)', function() {
    it('should respond with a 403', function (done) {
      var endpoint = '/user/' + user.id
      api.get(endpoint)
        .set('Accept',  'application/json')
        .set('Content-Type',  'application/json')
        .expect('Content-Type', /json/)
        .expect(403)
        .end(function(err, res) {
          if (err) throw err;
          done();
        });
    });
  });
```

Results:
```
 authenticated policy
    Unauthenticated (no token)
      1) should respond with a 403


  0 passing (1s)
  1 failing

  1) authenticated policy Unauthenticated (no token) should respond with a 403:
     Uncaught Error: expected "Content-Type" matching /json/, got "text/html; charset=utf-8"
    at net.js:1419:10
```

Changing `jsonx` to `json` in forbidden.js makes the test pass.

What's the difference between `jsonx` and `json` anyway?

Thanks

**Comment by sailsbot** [comment]
Thanks for posting, @brandonhesse.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
